src/useObserver.ts

```diff
@@ -49,16 +95,14 @@ export function useObserver<T>(fn: () => T, baseComponentName: string = "observe
 
         const trackingData = createTrackingData(newReaction)
         reactionTrackingRef.current = trackingData
-        scheduleCleanupOfReactionIfLeaked(reactionTrackingRef)
```

Comment:
In the endgame we will need to feature detect for FinalizationRegistry and use the prev impl as fallback
FinalizationRegistry is on by default in node 14, firefox, chrome, (and soonish safari?)